Debounce Library

penguineer opened the pull request:
Software-Entprell-Mechanismus nach http://hackaday.com/2015/12/10/embed-with-elliot-debounce-your-noisy-buttons-part-ii/

Getestet mit einem ATMega8 auf dem Steckbrett.

penguineer committed:
Gitignore for the debouncing part

penguineer committed:
Debounce "library"

Based on
http://hackaday.com/2015/12/10/embed-with-elliot-debounce-your-noisy-buttons-part-ii/

penguineer committed:
Debounce demo

LeSpocky commented:
Funktionell sieht's okay aus. Vom Style sieht es eher nach quick'n'dirty aus:
- trailing newlines
- Tabs und Spaces gemischt
- Klammern nicht einheitlich gesetzt (bspw. { mal in der selben Zeile, mal nicht, mal mit Leerzeichen mal nicht
- in debounce.c die Patterns als binäre Literale aber die History wird mit hexadezimalen zurückgesetzt
- in firmware.c die Kommentare auf deutsch
- für die Funktionsheader wurde doxygen style benutzt, für die file header nicht
- keine Lizenz angegeben
- etwas mehr Doku zur empfohlenen Update-Rate wäre gut
- es könnte ggf. auch stdbool.h eingebunden werden, damit die Funktionen debounce_is_\* vom Prototyp her erkennen lassen, was sie zurückgeben

penguineer commented:
Danke für das Feedback – schau ich mir diesbezüglich nochmal an.

penguineer committed:
Debounce library code quality

Add fixes regarding https://github.com/netz39/rollladensteuerung/pull/1#issuecomment-214857090

penguineer committed:
Debounce demo code quality

Add fixes regarding https://github.com/netz39/rollladensteuerung/pull/1#issuecomment-214857090

penguineer commented:
Bzgl. der Update-Rate kann ich nichts schreiben. Da muss man den Artikel lesen bzw. schauen, was in der Anwendung gut funktioniert.
